src/main/java/com/exasol/ReleaseRobot.java

```diff
@@ -1,88 +1,171 @@
             } else {
-                repositoryHandler.validate();
-                repositoryHandler.release();
+                runRelease(repositoryHandler);
             }
         } catch (final RuntimeException exception) {
-            LOGGER.severe(MessageFormat.format("'{}' request failed. Cause: {}", goalAsString, exception.getMessage()));
+            LOGGER.severe(() -> "'" + this.goal + "' request failed. Cause: " + exception.getMessage());
         }
     }
 
-    private RepositoryHandler getRepositoryHandler(final String repositoryName, final String[] platforms) {
-        final Set<ReleasePlatform> platformsList = ReleasePlatform.toSet(platforms);
-        final CredentialsProvider credentialsProvider = CredentialsProvider.getInstance();
-        final GitHubRepository repository = GitHubRepositoryFactory.getInstance().createGitHubRepository(
-                REPOSITORY_OWNER, repositoryName, credentialsProvider.provideGitHubCredentials());
-        return new RepositoryHandler(repository, platformsList);
+    private void runValidation(final RepositoryHandler repositoryHandler) {
+        if (this.gitBranch == null || this.gitBranch.isEmpty()) {
+            repositoryHandler.validate(Optional.empty());
+        } else {
+            repositoryHandler.validate(Optional.of(this.gitBranch));
+        }
+    }
+
+    private void runRelease(final RepositoryHandler repositoryHandler) {
+        repositoryHandler.validate(Optional.empty());
+        repositoryHandler.release();
     }
 
     /**
-     * Run the Release Robot.
-     * 
-     * @param args arguments
+     * Get a {@link ReleaseRobot} builder
+     *
+     * @return builder instance
      */
-    public static void main(final String[] args) {
-        final Options options = createOptions();
-        final CommandLine cmd = getCommandLine(args, options);
-        final String[] platformsArray = cmd.getOptionValue(PLATFORM_SHORT_OPTION).split(",");
-        final ReleaseRobot releaseRobot = new ReleaseRobot();
-        releaseRobot.dispatch(cmd.getOptionValue(NAME_SHORT_OPTION), cmd.getOptionValue(GOAL_SHORT_OPTION),
-                platformsArray);
+    public static Builder builder() {
+        return new Builder();
     }
 
-    private static Options createOptions() {
-        final Option name = new Option(NAME_SHORT_OPTION, "name", true, "project name");
-        name.setRequired(true);
-        final Option goal = new Option(GOAL_SHORT_OPTION, "goal", true, "goal to execute");
-        goal.setRequired(true);
-        final Option platforms = new Option(PLATFORM_SHORT_OPTION, "platforms", true,
-                "comma-separated list of release platforms");
-        platforms.setRequired(true);
-        return new Options().addOption(name).addOption(goal).addOption(platforms);
-    }
+    /**
+     * Builder for {@link ReleaseRobot}
+     */
+    public static final class Builder {
+        private String gitBranch;
```

Comment:
Holding platform specific values in the main class creates tight coupling. This way you block your way to making RR extensible.